cpu/cc2538/radio/cc2538_rf_radio_ops.c

```diff
@@ -226,8 +255,11 @@ static int _read(ieee802154_dev_t *dev, void *buf, size_t size, ieee802154_rx_in
     }
 
 end:
-    /* Enable RX Chain */
-    RFCORE_XREG_FRMCTRL0 &= ~CC2538_FRMCTRL0_RX_MODE_DIS;
+    /* Enable RX Chain only if the state is still RX_ON */
+    if (cc2538_state == CC2538_STATE_READY
+            || cc2538_state == CC2538_STATE_TX_ACK) {
```

Comment:
How do `CC2538_STATE_READY` and `CC2538_STATE_TX_ACK` correspond with RX_ON?

Reply:
I rewrote this logic to be more specific